task3.js

```diff
@@ -0,0 +1,53 @@
+// 3. Implementing Basic Back Pressure
+// Task: Implement a readable and a writable stream where the writable stream deliberately
+// writes data slower than the readable reads it, demonstrating how back pressure is managed.
+const { Readable, Writable } = require('stream');
+
+// CUSTOM READABLE CLASS
+class MyReadableFast extends Readable {
+  constructor() {
+    super();
+    this.currIndex = 0;
+  }
+
+  read(size) {
+    if (this.currIndex < 100) {
+      const chunk = `Chunk ${this.currIndex++}\n`;
+      console.log(`Pushing: ${chunk}`);
+      this.push(chunk);
+    } else {
+      this.push(null); 
+    }
+  }
+}
+
+// CUSTOM WRITABLE CLASS
+class MyWritableSlow extends Writable {
+  constructor() {
+    super();
+  }
+
+  write(chunk, coding, callback) {
+    console.log(`Writing: ${chunk.toString()}`);
+    setTimeout(callback, 500); 
+  }
+}
+
+const fastReadable = new MyReadableFast();
+const slowWritable = new MyWritableSlow();
+
+
+fastReadable.pipe(slowWritable);
```

Comment:
Streams handle backpressure automatically, let's use writable.writ() and check it's return value if it is false then we reached highwatermark and we should wait when drain event fires.